README-FRONT-END-DEV.md

```diff
@@ -0,0 +1,204 @@
+
+If you're looking for the technical specs of the Messenger service itself, you'll want the [README.md](README.md) document.
+
+Prerequisite: the assumption is that you're already familiar with Zooniverse front end development, e.g. you know how to work with the Zooniverse [starter project/template](https://github.com/zooniverse/zoo-reduxify/).
+
+## 0. Setup
+
+Here's a checklist of things to do before starting your Custom Front End project that accesses the Messenger service.
+
+* Are you clear whether your project is using `staging` or `production`?
+  * There are `staging` and `production` versions of Messenger, so make sure your choice of environment matches.
+  * Reminder: if your web-based CFE uses the [Panoptes Javascript Client](https://www.npmjs.com/package/panoptes-client) [(src)](https://github.com/zooniverse/panoptes-javascript-client/), you can switch enviornments by appending `?env=staging` to your URL.
+* Have you created a Panoptes App for your CFE?
+  * If not, go to [https://panoptes.zooniverse.org/oauth/applications](https://panoptes.zooniverse.org/oauth/applications) or [https://panoptes-staging.zooniverse.org/oauth/applications](https://panoptes-staging.zooniverse.org/oauth/applications).
+* Can you **login to your CFE using your Zooniverse account?** 
+  * i.e. Have you configured your CFE to connect to the Panoptes auth systems? Check your environment, Panoptes App ID, and the valid Callback URLs in the Panoptes OAuth2 Provider/Doorkeeper (links above).
+  * Reminder: on your first login to your CFE, you should be asked to _authorise the Panoptes App._
+* Are you the owner or collaborator of the Project?
+  * You cannot add Transcriptions to Subjects belonging to a Project where you're not an owner/collaborator.
+
+Be aware that the Messenger service has different URLs for its `staging` and `production` settings.
+* `staging` - `https://messenger-staging.zooniverse.org/`
+* `production` - TBC
+
+## 1. Register a Project
+
+Before you start adding Transcriptions to Subjects, you first need to register the Panoptes Project that the Subjects belong to.
+
+This uses the `POST /projects` functionality of the Messenger service, and should only be done **once.** Attempting to register the same project multiple times will net you a bunch of errors.
+
+Here's an example of registering (on `staging`) Project 1651, aka `darkeshard/transfromers`:
+
+```
+import apiClient from 'panoptes-client/lib/api-client.js';
+
+var url = 'https://messenger-staging.zooniverse.org/projects/' +
+          '?slug=' + encodeURIComponent('darkeshard/transformers');  //OK, honestly, I'm not sure if this ?slug part is necessary, but my personal notes says yes? (@shaun.a.noordin 20170630)
```

Comment:
There shouldn't be any query params on a POST request. The body should be formatted as indicated, but the `slug=[...]` bit will be ignored.